src/contribute/contribution-guide.md

```diff
@@ -201,10 +201,12 @@ For contributions to the Java code, run unit tests locally via Maven. Alternativ
 Once the initial code is complete and the tests pass, it’s time to start the code review process. We review and discuss all code, no matter who authors it. It’s a great way to build community, since you can learn from other developers, and they become familiar with your contribution. It also builds a strong project by encouraging a high quality bar and keeping code consistent throughout the project.
 
 ### Create a pull request
-Organize your commits to make your reviewer’s job easier. Use the following command to re-order, squash, edit, or change description of individual commits.
+Organize your commits to make your reviewer’s job easier. Reviewers normally prefer a minimum number of commits that break up a PR into logical steps. For most PRs, you'll squash your changes down to 1 commit. You can use the following command to re-order, squash, edit, or change description of individual commits.
 
     $ git rebase -i origin/master
 
+You'll then push to your branch on github. Note: when updating your commit after PR feedback and use squash to get back to one commit, you will need to do a force submit to the branch on your repo.
```

Comment:
Let's say I have a PR with 2 logical steps to it, changing a medium sized set of files. 

I break my initial PR up into 2 commits, each addressing a subset of the files. 

I then need to react to PR feedback. If I need to change the files in both commits, is there a way to correctly squash the changes so that the files I change are separated out into the correctly commits? 

If they don't squash the changes, then we get a bunch of commits like "updating based on PR feedback", so obviously that's not right either.

If there isn't a good way to do this, then inevitably 1 of the 2 commits ends up getting a bunch of the random fixes, right? So why shouldn't we just be telling people to do 1 commit?